packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -359,6 +385,55 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
             .map((fieldKey) => this.proxifyField(fieldKey, false))
             .values();
     }
+
+    public createField(fieldKey: FieldKey, newContent: ITreeCursor): EditableField | undefined {
+        assert(!this.has(fieldKey), "The field already exists.");
+        const fieldKind = this.lookupFieldKind(fieldKey);
+        assert(
+            fieldKind.multiplicity !== Multiplicity.Value,
+            "It is invalid to create fields of kind `value` as they should always exist.",
+        );
+        const path = this.cursor.getPath();
+        switch (fieldKind.multiplicity) {
+            case Multiplicity.Optional: {
+                if (this.context.setOptionalField(path, fieldKey, newContent, true))
+                    return this.proxifyField(fieldKey, false);
+            }
+            case Multiplicity.Sequence: {
+                if (this.context.insertNodes(path, fieldKey, 0, newContent))
+                    return this.proxifyField(fieldKey, false);
+            }
+            default:
+        }
+        return undefined;
+    }
+
+    public deleteField(fieldKey: FieldKey): void {
+        assert(this.has(fieldKey), "The field does not exist.");
+        const fieldKind = this.lookupFieldKind(fieldKey);
+        assert(
```

Comment:
We can do this same checking, but in a more maintainable (less redundant) and more performant way by moving the error to the default case in the switch statement (where it can call "fail").

Reply:
done